checkstyle.xml

```diff
@@ -0,0 +1,38 @@
+<?xml version="1.0" encoding="utf-8" ?>
+
+<!DOCTYPE module PUBLIC
+        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
+        "https://checkstyle.org/dtds/configuration_1_3.dtd">
+
+<module name="Checker">
+    <module name="TreeWalker">
+        <module name="ConstantName"/>
+
+        <!-- Validate that first sentence is not empty and first sentence is not missing -->
+        <module name="SummaryJavadocCheck"/>
+        <module name="InvalidJavadocPosition"/>
+
+        <module name="JavadocType">
+            <property name="scope" value="package"/>
+        </module>
+        <module name="MissingJavadocType">
+            <property name="scope" value="package"/>
+        </module>
+
+        <module name="JavadocMethod">
+            <property name="accessModifiers" value="public, package"/>
+            <property name="allowMissingParamTags" value="true"/>
+            <property name="allowMissingReturnTag" value="true"/>
+        </module>
+        <module name="MissingJavadocMethod">
+            <property name="scope" value="package"/>
+            <property name="excludeScope" value="protected"/>
```

Comment:
Is this really necessary? Since you specify which scope to include, also specifying an exclusion seems redundant.

Reply:
I initially thought I would exclude enforcing Javadoc for `protected` methods and that's why I added this property. Maybe I could include scope `package` in the `JavadocMethod` as that would ensure that if the Javadoc is present, it follows a format.

Reply:
You merged it so doesn't matter now, It was really trivial anyway. :sweat_smile:

Reply:
Yeah, this will include package private and public but exclude protected. Again, I didn't know that scopes were inclusive upward.

Anyway, the more relevant question here would probably have been _why_ one would exclude protected, seeing as it's a strictly more permissive scope than package private.

Reply:
I just thought that we rarely come across `protected` methods, so excluding the Javadoc for it is no harm. Including isn't either, though. I can change this when I come across such a method.